### PR TITLE
Create home page for an Apple site.

### PR DESCRIPTION
**Pull request by mbabaliiryn**
Create a home page for an Apple site

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+#nav-bar {
+  background-color: #565457;
+  transform: translateY(100px);
+  padding: 15px 0;
+  border-radius: 10px;
+  margin-left: 150px;
+  margin-right: 150px;
+  height: 30px;
+}
```
Comment (suretrust) [inline style.css]: #### Navigation background

* [ ] It uses CSS [`background -> gradient`](https://www.w3schools.com/cssref/css3_pr_background.asp) for background ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/building_with_backgrounds_and_gradients/navigation_requirement.png) from [Odin](https://www.theodinproject.com/courses/html5-and-css3/lessons/building-with-backgrounds-and-gradients))

File: index.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                <li><a href="#" class="nav-list">iPod</a></li>
+                <li><a href="#" class="nav-list">iPhone</a></li>
+                <li><a href="#" class="nav-list">iTunes</a></li>
+                <li><a href="#" class="nav-list">Support</a></li>
+                <li>
+                       <input id="input"/>
+                        <button><i id="search-icon" class="fas fa-search"></i></button>
+                </li>  
+            </ul>
+        </nav>
```
Comment (suretrust) [inline index.html]: Your navigation bar doesn't look close to the original with the right gradients.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        <h1 class="text"><span>What will your verse be?</span><br><br>Everyone has something to share.see how people are using iPads in their unique ways to add the world's story.<br><br>Watch the "Your Verse" film and explore the stories</h1> 
+    </div>
+        </main>
+    <footer>
+        <div id="img">
+            <img class="img2" src="img/promo_iphone5s.jpg" alt="">
+            <img class="img2" src="img/promo_iphone5c.jpg" alt="">
+            <img class="img2" src="img/promo_30_years.jpg" alt="">
+            <img class="img2" src="img/promo_supplier_responsibility.jpg" alt="">
+    </div>
```
Comment (suretrust) [inline index.html]: These images are overflowing, ensure each has a 25% width relative to the screen width.
Screenshot 2020-01-22 at 06 45 51

File: index.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        <div id="main-paragraph">
+        <h1 class="text"><span>What will your verse be?</span><br><br>Everyone has something to share.see how people are using iPads in their unique ways to add the world's story.<br><br>Watch the "Your Verse" film and explore the stories</h1> 
+    </div>
+        </main>
+    <footer>
+        <div id="img">
+            <img class="img2" src="img/promo_iphone5s.jpg" alt="">
+            <img class="img2" src="img/promo_iphone5c.jpg" alt="">
+            <img class="img2" src="img/promo_30_years.jpg" alt="">
+            <img class="img2" src="img/promo_supplier_responsibility.jpg" alt="">
```
Comment (suretrust) [inline index.html]: Each of these images should be wrapped in an `a` tag.

#### Bottom

* [ ] It contains four elements which are `links` (to another articles or `#`) ([link to page](https://web.archive.org/web/20140301004610/http://www.apple.com/))

**Comment by mbabaliiryn** [comment]
Hello Saheed Oladele,
Thanks for viewing my code. I have implemented the required changes.

NANYONJO IRENE,
0758323723,
mbabaliiryn@gmail.com,
Kisaasi - Kampala.

**Comment by fatymahmed** [comment]
## Hi @mbabaliiryn,
Great job 👏👏 on making some of the changes requested by the previous TSE like adding a background gradient to your navbar,however, for the project to be approved, you need to make all of them unless they are stated as optional.

The following are changes you should make as suggested by the previous TSE

> #### Horizontal Scroll
> You seem to have a little bug with the footer of the page, shooting out beyond the browser's maximum width. Please fix this.
- The section above the footer displays three images in a row and the next one is pushed below the three images, please fix this so that all four images should be on the same row as it is on the apple page.
> screenshot
> 
> #### Hero Section Font Sizing
> 
> Please consider reducing the `font-size` of the text in your hero section to within appropriate range. A place way to start is to inspect the source code on the original site to give you an idea of what the right value may be. Please give this some thought as it will give your project some more shine.
> screenshot
> #### Descriptive PR
> 
> You need a more descriptive PR as it is a hard requirement. Consider describing the work you have completed in the branch in as much detail as you can and even include a link to the live project to speed up your review!
> screenshot
> #### Descriptive Readme
- Great job updating the README file, please update the links in your file so that they are not local links like the ones you currently have.
You can use [this](https://raw.githack.com/) to generate a live link and use it in your README file.

File: style.css [hunk 5]
```diff
@@ -0,0 +1,117 @@
+* {
+  margin: 0;
+  padding: 0;
+}
+
+body {
+  overflow: scroll;
+}
```
Comment (suretrust) [inline style.css]: We don't want any overflow. This is one of the reasons why your page is overflowing.

File: style.css [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+#nav-bar {
+  background: linear-gradient(90deg, rgba(2, 0, 36, 1) 0%, rgba(75, 75, 108, 0.9598214285714286) 52%, rgba(130, 125, 125, 0.8841911764705882) 100%);
+  transform: translateY(100px);
+  padding: 15px 0;
+  border-radius: 10px;
+  margin-left: 150px;
+  margin-right: 150px;
+  height: 30px;
+  border: 1px solid #5d5d61;
+}
```
Comment (suretrust) [inline style.css]: The gradient ought to be down-up/up-down, not left-right/right-left. Please see the original page for clarification.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        <h1 class="text"><span>What will your verse be?</span><br><br>Everyone has something to share.see how people are using iPads in their unique ways to add the world's story.<br><br>Watch the "Your Verse" film and explore the stories</h1> 
+    </div>
+        </main>
+    <footer>
+        <div id="img">
+            <a href="#"><img class="img2" src="img/promo_iphone5s.jpg" alt="promo_iphone5c"></a>
+            <a href="#"><img class="img2" src="img/promo_iphone5c.jpg" alt="promo_iphone5s"></a>
+            <a href="#"><img class="img2" src="img/promo_30_years.jpg" alt="promo_30_years"></a>
+            <a href="#"><img class="img2" src="img/promo_supplier_responsibility.jpg" alt="promo_supplier_responsibility"></a>
+    </div>
```
Comment (suretrust) [inline index.html]: These are still not well arranged on the same line. The code below should fix this. We set a width of 100 view-width on the parent image and set 25% for each `a` tag inside of the `#img` element. Please try to make this work, you are almost there.

```css
#img {
  width: 100vw;
}

#img a {
  width: 25%;
}
```

**Comment by euqueme** [comment]
# Hello @mbabaliiryn :bulb: 

## First of all: Great job so far and thank you for doing all the changes the PR's are giving you :medal_sports: There are still a few Mandatory things to fix but I can see you already have come a long way :muscle: 

**Navbar**

- [ ] It should be made of nine different buttons with the Apple logo on the leftmost section and a search field on the rightmost section (you are missing the ipad button)

image

- [ ] Your Bottom images don't fit in big screens (make them 24.7% wide instead of 24.8%)

image

- [ ] Don't use \<div\> and \<span\> tags since these are non-semantic elements as stated in the [Additional Microverse Requirements](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59538830/) in the section (Remember to use semantic HTML.)

**From the Odin Project**

- [ ] Set up the main background image. You can grab that exact image from their webpage by finding the \<section class=\"hero\"\> tag in your inspector and checking out its background [image](https://web.archive.org/web/20140228114747im_/http://images.apple.com/v/home/as/images/your_verse_hero.jpg).

- [ ] Fix these links in your Readme file, the URLs lead to non-existing pages

image

image

[OPTIONAL] I really encourage you to try to match the font size of the main text and the footer as closer to the original as it can be.

Yours:

image

image


Original:

image

image

Reach to me in slack (@ Maru) if you have any questions (looking forward to hear from you :raised_hands: )

## That's it. Please :pray: correct these changes and submit back for a code review when you are done :rocket:

**Comment by mbabaliiryn** [comment]
Thank you for your feedback but the image in the original doesn't fit well
in the background
NANYONJO IRENE,
0758323723,
mbabaliiryn@gmail.com,
Kisaasi - Kampala.

**Comment by euqueme** [comment]
> Thank you for your feedback but the image in the original doesn't fit well in the background NANYONJO IRENE, 0758323723, mbabaliiryn@gmail.com, Kisaasi - Kampala.

To make the image fit in the background you can change these settings in your CSS file:

```
    height: 550px;
    background: url(img/hero.jpg) 50% 100% no-repeat;
```

image

**Comment by euqueme** [comment]
> **Navbar**
> 
> * [ ]  It should be made of nine different buttons with the Apple logo on the leftmost section and a search field on the rightmost section (you are missing the ipad button)
> 
> image
> 
> * [ ]  Your Bottom images don't fit in big screens (make them 24.7% wide instead of 24.8%)
> 
> image
> 
> * [ ]  Don't use <div> and <span> tags since these are non-semantic elements as stated in the [Additional Microverse Requirements](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59538830/) in the section (Remember to use semantic HTML.)
> 
> **From the Odin Project**
> 
> * [ ]  Set up the main background image. You can grab that exact image from their webpage by finding the <section class="hero"> tag in your inspector and checking out its background [image](https://web.archive.org/web/20140228114747im_/http://images.apple.com/v/home/as/images/your_verse_hero.jpg).
> * [ ]  Fix these links in your Readme file, the URLs lead to non-existing pages
> 
> image
> 
> image
> 
> **Reach to me in slack (@ Maru) if you have any questions (looking forward to hear from you  )**
> 

# Please fix the Mandatory Requirements Listed Above :arrow_up: :arrow_up:  before submitting for a code review again

**Comment by mbabaliiryn** [comment]
still it can not fit on my screen
NANYONJO IRENE,
0758323723,
mbabaliiryn@gmail.com,
Kisaasi - Kampala.

**Comment by Kyle-Law** [comment]
Review from your coding partner - @Kyle-Law : 
Hi @mbabaliiryn ! Here is a review regarding your bottom images alignment:
1) Your `footer #img a` styling are not applied because of the wrong selector. Change it to `footer #img a img` to select the images. 
2) Then, adjust the width to around 24.5%~24.7% to align 4 images in 1 row.

**Comment by Kyle-Law** [comment]
Great, your footer images are now aligned. 
Next, replace your README live link to this: 
https://rawcdn.githack.com/mbabaliiryn/Building-with-Backgrounds-and-Gradients/68f9655a829fb871f4e2fd0ecbaf049a9b472cc0/index.html

File: style.css [hunk 8]
```diff
@@ -0,0 +1,118 @@
+* {
+  margin: 0;
+  padding: 0;
+  outline: none;
+}
+
+main {
+  background-image: url("img/large.jpg");
```
Comment (euqueme) [inline style.css]: Replace this line with the following:
background: url(img/hero.jpg) 50% 100% no-repeat;

File: style.css [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  background-repeat: no-repeat;
```
Comment (euqueme) [inline style.css]: delete this line

File: style.css [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+  background-size: cover;
```
Comment (euqueme) [inline style.css]: delete this line

File: style.css [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+  height: 800px;
```
Comment (euqueme) [inline style.css]: replace this line with:
height: 525px;

File: Readme.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+Home page for an Apple site
+http://127.0.0.1:5500/index.html
+
+The aim of the project is to build backgrounds and gradients.The project contains a nav-bar, main-section plus the footer
+
+# Built with
+HTML and CSS3
+
+# Live demo
+.[Live demo link].http://127.0.0.1:5502/index.html
```
Comment (euqueme) [inline Readme.md]: Replace this line with a link that takes you to the following URL:
https://rawcdn.githack.com/mbabaliiryn/Building-with-Backgrounds-and-Gradients/68f9655a829fb871f4e2fd0ecbaf049a9b472cc0/index.html

File: Readme.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
+# Building-with-backgrounds and Gradients
+Home page for an Apple site
+http://127.0.0.1:5500/index.html
```
Comment (euqueme) [inline Readme.md]: Replace this line with a link that takes you to this URL:
https://web.archive.org/web/20140301004610/http://www.apple.com/

**Comment by mbabaliiryn** [comment]
Thank you so much Maru I have done the changes
NANYONJO IRENE,
0758323723,
mbabaliiryn@gmail.com,
Kisaasi - Kampala.

**Pull request merged**